pkg/controller/toolchainstatus/toolchainstatus_controller.go

```diff
@@ -399,8 +396,8 @@ func compareAndAssignMemberStatuses(reqLogger logr.Logger, toolchainStatus *tool
 		if ok {
 			toolchainStatus.Status.Members[index].MemberStatus = newMemberStatus
 			delete(members, memberStatus.ClusterName)
-		} else if memberStatus.ClusterName != "" && memberStatus.UserAccountCount > 0 {
-			err := fmt.Errorf("toolchainCluster not found for member cluster %s that was previously registered in the host", memberStatus.ClusterName)
+		} else if memberStatus.UserAccountCount > 0 {
+			err := fmt.Errorf("ToolchainCluster CR wasn't found for member cluster `%s` that was previously registered in the host", memberStatus.ClusterName)
 			reqLogger.Error(err, "the member cluster seems to be removed")
 			memberStatusNotFoundCondition := status.NewComponentErrorCondition(toolchainv1alpha1.ToolchainStatusMemberToolchainClusterRemovedReason, err.Error())
```

Comment:
So we don't really have a removed condition anymore right? Maybe we should change `ToolchainStatusMemberToolchainClusterRemovedReason` to be `ToolchainStatusMemberToolchainClusterMissingReason`?

Reply:
> So we don't really have a removed condition anymore right?

It's not a condition but a reason, but I probably don't understand the question. We still use it as you can see in the code.

>  Maybe we should change ToolchainStatusMemberToolchainClusterRemovedReason to be ToolchainStatusMemberToolchainClusterMissingReason?

yes, that would probably make more sense

Reply:
Sorry, I probably could have worded that better. I didn't see `ToolchainStatusMemberToolchainClusterRemovedReason` used anywhere else. I mainly just meant we could rename the reason to `ToolchainStatusMemberToolchainClusterMissingReason` since with the changed approach we aren't going to have a condition in the Members section for a cluster that is removed successfully.

Reply:
Didn't you mean that we can remove `ToolchainStatusMemberStatusNoClustersFoundReason`? And just rename the `ToolchainStatusMemberToolchainClusterRemovedReason` to `ToolchainStatusMemberToolchainClusterMissingReason`?
I'll do that

Reply:
https://github.com/codeready-toolchain/api/pull/207